sklearn/linear_model/_logistic.py

```diff
@@ -751,14 +752,24 @@ def grad(x, *args): return _logistic_loss_and_grad(x, *args)[1]
     coefs = list()
     n_iter = np.zeros(len(Cs), dtype=np.int32)
     for i, C in enumerate(Cs):
-        if solver == 'lbfgs':
+        if solver in ['lbfgs', 'trust-ncg']:
             iprint = [-1, 50, 1, 100, 101][
                 np.searchsorted(np.array([0, 1, 2, 3]), verbose)]
-            opt_res = optimize.minimize(
-                func, w0, method="L-BFGS-B", jac=True,
-                args=(X, target, 1. / C, sample_weight),
-                options={"iprint": iprint, "gtol": tol, "maxiter": max_iter}
-            )
+            if solver == 'lbfgs' or multi_class != "multinomial":
```

Comment:
Unless there is an issue with `multi_class != "multinomial"` and `solver=trust-ncg`, we should keep this open and add the `hess` for  'trust-ncg' in the `multi_class != "multinomial"` case.

In other words, if a user set `solver='trust-ncg'` and `multi_class='ovr'`, then the current code will use `solver='lbfgs'`